### PR TITLE
Add `hydro.mux` operator and initial API tests

### PR DESCRIPTION
**Pull request by shadaj**
Add `hydro.mux` operator and initial API tests

### CODE REVIEW

**Comment by shadaj** [comment]
More importantly, adds an initial set of Python API tests so I stop introducing regressions

File: hydro_cli/src/core/hydroflow_crate/ports.rs [hunk 1]
```diff
@@ -156,12 +199,54 @@ impl HydroflowPortConfig {
...
+    fn record_server_config(&mut self, config: ServerConfig) {
+        let from = self.service.upgrade().unwrap();
         let mut from_write = from.try_write().unwrap();
-        from_write
-            .add_connection(&from, self.port.clone(), sink)
-            .unwrap();
+
+        // TODO(shadaj): if already in this map, we want to broadcast
+        assert!(!from_write.port_to_server.contains_key(&self.port));
+        from_write.port_to_server.insert(self.port.clone(), config);
```
Comment (MingweiSamuel) [inline hydro_cli/src/core/hydroflow_crate/ports.rs:249]: todo?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -170,6 +170,39 @@ jobs:
...
+          pip install maturin
+          maturin develop
+
+      - name: Run Python tests
+        run: |
+          cd hydro_cli
+          source .venv/bin/activate
+          cd python_tests
+          pip install -r requirements.txt
+          rustup run nightly-2023-03-01-x86_64-unknown-linux-gnu pytest *.py
```
Comment (MingweiSamuel) [inline .github/workflows/ci.yml:204]: Could we do something like
```bash
rustup run "$(grep -oP '(?<=channel = ")\S+(?=")' rust-toolchain.toml)-x86_64-unknown-linux-gnu" pytest *.py
```

File: rust-toolchain.toml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2023-03-01"
+channel = "nightly-2023-03-01" # make sure to update ci.yml/test-cli!
```
Comment (MingweiSamuel) [inline rust-toolchain.toml:2]: ^

**Review by MingweiSamuel — APPROVED** [review]

File: hydro_cli/src/core/hydroflow_crate/ports.rs (hunk 1 above)
Comment (shadaj) [reply hydro_cli/src/core/hydroflow_crate/ports.rs:249]: same TODO as elsewhere, need to address in a separate PR though...

**Pull request merged**